receiver/mongodbatlasreceiver/logs.go

```diff
@@ -0,0 +1,366 @@
+	}
+
+	// create an array with a comma delimiter from the original string
+	tmp := strings.Split(s, ",")
+	for _, t := range tmp {
+
+		u, err := url.Parse(t)
+		if err != nil {
+			fmt.Printf("Error parsing out %s", t)
+		}
+
+		// separate hostname from scheme and port
+		host, _, err := net.SplitHostPort(u.Host)
+		if err != nil {
+			// the scheme prefix was not added on to this string
+			// thus the hostname will have been placed under the "Scheme" field
+			hostnames = append(hostnames, u.Scheme)
+		} else {
+			// hostname parsed successfully
+			hostnames = append(hostnames, host)
+		}
+
+	}
+
+	return hostnames
+}
+
+// KickoffReceiver spins off functionality of the receiver from the Start function
+func (s *receiver) KickoffReceiver(ctx context.Context) {
+	resource := resourceInfo{start: strconv.Itoa(int(time.Now().Unix())), end: ""}
+	cfgProjects, err := s.createProjectMap()
+	if err != nil {
+		s.log.Error("Failed to create project map", zap.Error(err))
+	}
+
+	for {
+		// reterive all organizations listed in the MongoDB client
+		orgs, err := s.client.Organizations(ctx)
+		if err != nil {
+			s.log.Error("Error retrieving organizations", zap.Error(err))
+		}
+		// get all projects listed for each of the organizations
+		for _, org := range orgs {
+			projects, err := s.client.Projects(ctx, org.ID)
+			if err != nil {
+				s.log.Error("Error retrieving projects", zap.Error(err))
+			}
+
+			// filter out any projects not specified in the config
+			fp := filterProjects(projects, cfgProjects)
+
+			if len(projects) < 1 {
+				return
+			}
+
+			// get clusters for each of the projects
+			for _, project := range fp {
+				resource.Org, resource.Project = *org, *project
+				clusters, err := s.processClusters(ctx, cfgProjects, &resource)
+				if err != nil {
+					s.log.Error("Failure to process Clusters", zap.Error(err))
+				}
+
+				if resource.end != "" {
+					resource.start = resource.end
+				}
+				resource.end = strconv.Itoa(int(time.Now().Unix()))
+				s.collectClusterLogs(clusters, cfgProjects, resource)
+			}
+		}
+
+		// collection interval loop,
+		select {
+		case <-ctx.Done():
+			return
+		case <-s.stopperChan:
+			return
+		case <-time.After(collectionInterval):
+
+		}
+	}
+}
+
+func filterProjects(projects []*mongodbatlas.Project, cfgProjects map[string]*Project) map[string]*mongodbatlas.Project {
+	fp := make(map[string]*mongodbatlas.Project)
+	for _, project := range projects {
+		if _, ok := cfgProjects[project.Name]; ok {
+			fp[project.Name] = project
+		}
+	}
+
+	return fp
+}
+
+func (s *receiver) processClusters(ctx context.Context, cfgProjects map[string]*Project, r *resourceInfo) ([]mongodbatlas.Cluster, error) {
+	include, exclude := cfgProjects[r.Project.Name].IncludeClusters, cfgProjects[r.Project.Name].ExcludeClusters
+	clusters, err := s.client.GetClusters(ctx, r.Project.ID)
+	if err != nil {
+		s.log.Error("Failure to collect clusters from project: %w", zap.Error(err))
+		return clusters, err
+	}
```

Comment:
Is it possible to get a partial response here? Like, can GetClusters return an error and some clusters? If it can, you should filter those clusters. If it cannot, you should just explicitly return nil.